src/util/scrollbar.ts

```diff
@@ -0,0 +1,44 @@
+import {Injectable, Inject} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+
+
+
+@Injectable()
+export class ScrollBar {
+  constructor(@Inject(DOCUMENT) private _document) {}
+
+  compensate() {
+    if (!this._isPresent()) {
+      return () => {};
+    }
+    const width = this._getWidth();
+    return this._adjustBody(width);
+  }
+
+  private _adjustBody(width: number) {
+    const {body} = this._document;
+    const userSetPadding = body.style.paddingRight;
+    const paddingAmount = parseFloat(window.getComputedStyle(body)['padding-right']);
```

Comment:
Hmm, I was just fixing SSR for the carousel and realised that `window` might break it too...

Reply:
Regardless, I don't think the fix for this should be a part of this PR